### PR TITLE
Popup stays open when tooltip is hovered onto

### PR DESCRIPTION
**Pull request by corrigancd**
Popup stays open when tooltip is hovered onto

PopupNowStaysOpen

### CODE REVIEW

File: public/vislib/marker_types/base_marker.js [hunk 1]
```diff
@@ -161,35 +161,32 @@ define(function (require) {
      */
     BaseMarker.prototype.bindPopup = function (feature, layer) {
       const self = this;
+      const KEEP_POPUP_OPEN_CLASS_NAMES = ['leaflet-popup', 'tooltip'];
+      self._getParent = function (element, classNames) {
```
Comment (szydan) [inline public/vislib/marker_types/base_marker.js]: please move out, does not have to be on self

File: public/tooltip/visTooltip.js [hunk 2]
```diff
@@ -85,9 +85,10 @@ define(function (require) {
               self.sirenMeta.vis.panelIndex
             );
 
-            //if vis is not assigned in coat tree, assign popup to main node
-            if (!etmVisNode) {
-              const mainNode = findMainCoatNode(sirenMetaTooltip.coat.items);
+            const mainNode = findMainCoatNode(sirenMetaTooltip.coat.items);
+            //if vis is not assigned in coat tree AND
+            //there is an search assigned to dashboard (i.e. main node)
+            if (!etmVisNode && mainNode) {
```
Comment (corrigancd) [inline public/tooltip/visTooltip.js]: @szydan, please check. The logic assigns the visualization in the map popup to the coat. If it is not assigned in the node, then we add the vis to the main node. 

I noticed that if a dashboard is not linked with any saved search, mainNode will  be undefined? Is this expected?

Comment (szydan) [reply public/tooltip/visTooltip.js]: yes 
no linked search === no main search

File: public/utils.js [hunk 3]
```diff
@@ -3,6 +3,25 @@ import _ from 'lodash';
...
+     * @param scale {parentClassNames} An array of classnames to check ancestory for
+     * @return {boolean} when true, the element has a parent whose
+     * class is specified in the paretnClassNames Array
+     */
+    getParent: function (element, parentClassNames) {
+      let parent = element;
+      while (parent != null) {
+        if (parent.className) {
+          for (let i = 0; i < parentClassNames.length; i++) {
+            if (L.DomUtil.hasClass(parent, parentClassNames[i])) return parent;
```
Comment (szydan) [inline public/utils.js]: here you are breaking the contract 
In the documentation you wrote 
```
@return {boolean} when true, the element has a parent whose
     * class is specified in the paretnClassNames Array
```
please return boolean or change the documentation

File: public/utils.js [hunk 4]
```diff
@@ -3,6 +3,25 @@ import _ from 'lodash';
 define(function (require) {
   const L = require('leaflet');
   return {
+    /*
+     * @param element {html element} The child element to check if there is a parent for
+     * @param scale {parentClassNames} An array of classnames to check ancestory for
+     * @return {object || boolean} when an object is returned, the element has a parent whose
```
Comment (szydan) [inline public/utils.js]: OK so if you actually want an object then I would change it to 
return and object if ... and undefined if not ...

Then simply remove the last line 
```
return false
```

**Comment by szydan** [comment]
@corrigancd 
I think the changes were also present in the other PR 
just to be sure please rebase on master and see if there is still any change
If not we can just close this PR

**Comment by corrigancd** [comment]
I rebased and tidied up some other things.

**Pull request merged**